edc-tests/edc-dataplane/cloud-transfer-tests/src/test/java/org/eclipse/tractusx/edc/dataplane/transfer/test/AzureToAzureTest.java

```diff
@@ -0,0 +1,244 @@
+
+        await().pollInterval(Duration.ofSeconds(10))
+                .atMost(Duration.ofSeconds(120))
+                .untilAsserted(() -> assertThat(consumerBlobHelper.listBlobs(CONSUMER_CONTAINER_NAME))
+                        .isNotEmpty()
+                        .contains(blobName)
+                        .contains(COMPLETION_MARKER));
+
+    }
+
+    @Test
+    void transferFile_targetContainerNotExist_shouldFail() {
+        // upload file to provider's blob store
+        var bcc = providerBlobHelper.createContainer(PROVIDER_CONTAINER_NAME);
+        providerBlobHelper.uploadBlob(bcc, TESTFILE_NAME);
+
+        // do NOT create container in consumer's blob store
+
+        DATAPLANE_RUNTIME.getVault().storeSecret(PROVIDER_KEY_ALIAS, PROVIDER_ACCOUNT_KEY);
+        DATAPLANE_RUNTIME.getVault().storeSecret(CONSUMER_KEY_ALIAS, """
+                {"sas": "%s","edctype":"dataspaceconnector:azuretoken"}
+                """.formatted(consumerBlobHelper.generateAccountSas(CONSUMER_CONTAINER_NAME)));
+
+        var request = createFlowRequest(TESTFILE_NAME);
+
+        var url = "http://localhost:%s/control/transfer".formatted(PROVIDER_CONTROL_PORT);
+
+        given()
+                .when()
+                .baseUri(url)
+                .contentType(ContentType.JSON)
+                .body(request)
+                .post()
+                .then()
+                .statusCode(200);
+
+        // wait until the data plane logs an exception that it cannot transfer the blob
+        await().pollInterval(Duration.ofSeconds(2))
+                .atMost(Duration.ofSeconds(10))
+                .untilAsserted(() -> verify(DATAPLANE_RUNTIME.getContext().getMonitor()).severe(eq("Error creating blob for %s on account %s".formatted(TESTFILE_NAME, CONSUMER_ACCOUNT_NAME)), isA(IOException.class)));
+    }
+
+    private DataFlowRequest createFlowRequest(String blobName) {
+        return DataFlowRequest.Builder.newInstance()
+                .id("test-request")
+                .sourceDataAddress(DataAddress.Builder.newInstance()
+                        .type("AzureStorage")
+                        .property("container", PROVIDER_CONTAINER_NAME)
+                        .property("account", PROVIDER_ACCOUNT_NAME)
+                        .property("keyName", PROVIDER_KEY_ALIAS)
+                        .property("blobname", blobName)
+                        .build()
+                )
+                .destinationDataAddress(DataAddress.Builder.newInstance()
+                        .type("AzureStorage")
+                        .property("container", CONSUMER_CONTAINER_NAME)
+                        .property("account", CONSUMER_ACCOUNT_NAME)
+                        .property("blobname", blobName)
+                        .property("keyName", CONSUMER_KEY_ALIAS)
+                        .build()
+                )
+                .processId("test-process-id")
+                .trackable(false)
+                .build();
+    }
```

Comment:
Pre-build `DataAdress` for Azure, S3 and later HTTP could remove a lot of duplicate code.
Can be part of a general refactoring Issue including the cosntants.